doc/sonic-sub-port-intf-hld.md

```diff
@@ -0,0 +1,475 @@
+|--------------------------------------------------------------------------------------------------------|
+| Verify that sub port interface mtu change is pushed to CONIFG_DB DOT1Q_INTERFACE table                 |
+| Verify that sub port interface mtu change is synced to APPL_DB INTF_TABLE by Intfmgrd                  |
+| Verify that sub port router interface entry in ASIC_DB has the updated mtu value                       |
+
+## 6.3 Sub port interface removal
+### 6.3.1 Remove an IP address from a sub port interface
+Test shall cover the IP address being an IPv4 address or an IPv6 address.
+
+| Test case description                                                                                  |
+|--------------------------------------------------------------------------------------------------------|
+| Verify that ip address configuration is removed from CONIFG_DB DOT1Q_INTERFACE table                   |
+| Verify that ip address configuration is removed from APPL_DB INTF_TABLE by Intfmgrd                    |
+| Verify that ip address state ok is removed from STATE_DB INTERFACE_TABLE by Intfmgrd                   |
+| Verify that subnet route entry is removed from ASIC_DB                                                 |
+| Verify that ip2me route entry is removed from ASIC_DB                                                  |
+
+### 6.3.2 Remove all IP addresses from a sub port interface
+| Test case description                                                                                  |
+|--------------------------------------------------------------------------------------------------------|
+| Verify that sub port router interface entry is removed from ASIC_DB                                    |
+
+### 6.3.3 Remove a sub port interface
+Test shall cover the parent interface being a physical port or a port channel.
+
+| Test case description                                                                                  |
+|--------------------------------------------------------------------------------------------------------|
+| Verify that sub port interface configuration is removed from CONIFG_DB DOT1Q_INTERFACE table           |
+| Verify that sub port interface configuration is removed from APPL_DB INTF_TABLE by Intfmgrd            |
+| Verify that sub port interface state ok is removed from STATE_DB by Intfmgrd                           |
+
+# 7 Scalability
+Scalability is ASIC-dependent.
+We enforce a minimum scalability requirement on the number of sub port interfaces that shall be supported on a SONiC switch.
+
+| Name                                                              | Scaling value             |
+|-------------------------------------------------------------------|---------------------------|
+| Number of sub port interfaces per phyical port or port channel    | 250                       |
+| Number of sub port interfaces per switch                          | 750                       |
```

Comment:
Why limit this? We have fixed number of port, and max `Number of sub port interfaces per phyical port or port channel` above, so we can get it by multiplication.

Reply:
Sub port interfaces may not be uniform across physical port and/or port channels. This is a number referred from Vendor manual. https://github.com/Azure/SONiC/pull/420#discussion_r303186368